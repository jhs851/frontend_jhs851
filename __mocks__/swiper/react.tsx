import React from 'react'

const mock: React.FC = ({ children }) => <div>{children}</div>

export default mock
