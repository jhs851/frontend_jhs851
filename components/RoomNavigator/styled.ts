import styled from 'styled-components'
import { motion } from 'framer-motion'

export const IconsContainer = styled(motion.div)`
  display: flex;
  flex-direction: row;
`

export const UploadIcon = styled.button`
  margin-right: 16px;
`
