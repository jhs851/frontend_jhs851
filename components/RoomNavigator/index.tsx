import React from 'react'
import Text from 'components/Text'
import Image from 'components/Image'
import { IconsContainer, UploadIcon } from './styled'
import Navigator from 'components/Navigator'
import { useRouter } from 'next/router'
import { motion } from 'framer-motion'
import MotionBackSVG from 'components/MotionBackSVG'
import useHeaderLeftVariants from './hooks/useHeaderLeftVariants'
import useHeaderRightVariants from './hooks/useHeaderLeftVariants'

type P = {
  roomName: string
  onToggleAssets: () => void
}

const RoomNavigator: React.FC<P> = ({ roomName, onToggleAssets }) => {
  const router = useRouter()
  const goBack = () => router.back()
  const onSearch = () => null
  const headerLeftVariants = useHeaderLeftVariants()
  const headerRightVariants = useHeaderRightVariants()

  return (
    <Navigator
      title={
        <motion.div {...headerLeftVariants}>
          <Text size={17} weight={700} letterSpacing={-0.12} color="white">
            {roomName}
          </Text>
        </motion.div>
      }
      headerLeft={
        <button data-testid="back-button" onClick={goBack}>
          <MotionBackSVG />
        </button>
      }
      headerRight={
        <IconsContainer {...headerRightVariants}>
          <UploadIcon data-testid="upload-button" onClick={onToggleAssets}>
            <Image
              src="/icons/upload.svg"
              alt="사진 업로드"
              width={24}
              height={24}
            />
          </UploadIcon>
          <button onClick={onSearch}>
            <Image src="/icons/search.svg" alt="검색" width={24} height={24} />
          </button>
        </IconsContainer>
      }
    />
  )
}

export default RoomNavigator
