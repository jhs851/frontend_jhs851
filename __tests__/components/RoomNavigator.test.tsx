import { fireEvent, render } from '@testing-library/react'
import RoomNavigator from 'components/RoomNavigator'

const useRouter = jest.spyOn(require('next/router'), 'useRouter')

describe('RoomNavigator 컴포넌트', () => {
  const onToggleAssets = jest.fn()
  const rendered = (roomName: string = '채팅방') =>
    render(
      <RoomNavigator roomName={roomName} onToggleAssets={onToggleAssets} />,
    )

  it('렌더링이 올바르게 된다', () => {
    expect(rendered().container).toMatchSnapshot()
  })

  it('뒤로가기 버튼을 누르면 router의 back 함수가 호출된다', () => {
    const back = jest.fn()
    useRouter.mockImplementationOnce(() => ({
      route: '/',
      pathname: '/',
      query: '',
      asPath: '',
      back,
    }))

    const { getByTestId } = rendered()

    fireEvent.click(getByTestId('back-button'))
    expect(back).toHaveBeenCalled()
  })

  it('업로드 아이콘을 클릭하면 onToggleAssets 함수가 호출된다', () => {
    const { getByTestId } = rendered()

    fireEvent.click(getByTestId('upload-button'))
    expect(onToggleAssets).toHaveBeenCalled()
  })
})
