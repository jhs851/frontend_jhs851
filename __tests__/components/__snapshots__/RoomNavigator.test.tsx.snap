// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomNavigator 컴포넌트 렌더링이 올바르게 된다 1`] = `
.c2 {
  font-family: AppleSDGothicNeo,sans-serif;
  font-size: 17px;
  font-weight: 700;
  -webkit-letter-spacing: -0.12px;
  -moz-letter-spacing: -0.12px;
  -ms-letter-spacing: -0.12px;
  letter-spacing: -0.12px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c5 {
  margin-right: 16px;
}

.c0 {
  height: 44px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  position: absolute;
  left: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

.c3 {
  position: absolute;
  right: 12px;
  top: 50%;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        data-testid="back-button"
      >
        <svg
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <line
            stroke="#fff"
            stroke-miterlimit="10"
            stroke-width="2"
            x1="16.2"
            x2="7.7"
            y1="19.8"
            y2="11.3"
          />
          <line
            stroke="#fff"
            stroke-miterlimit="10"
            stroke-width="2"
            x1="16.3"
            x2="7.7"
            y1="4.2"
            y2="12.6"
          />
        </svg>
      </button>
    </div>
    <div
      style="opacity: 0; transform: translateX(30px) translateZ(0);"
    >
      <p
        class="c2"
        color="white"
        letter-spacing="-0.12"
        size="17"
      >
        채팅방
      </p>
    </div>
    <div
      class="c3"
    >
      <div
        class="c4"
        style="opacity: 0; transform: translateX(30px) translateZ(0);"
      >
        <button
          class="c5"
          data-testid="upload-button"
        >
          <div
            class="c6"
          />
        </button>
        <button>
          <div
            class="c6"
          />
        </button>
      </div>
    </div>
  </header>
</div>
`;
